internal/common/conversion/type_conversion.go

```diff
@@ -56,3 +56,17 @@ func IsStringPresent(strPtr *string) bool {
 func MongoDBRegionToAWSRegion(region string) string {
 	return strings.ReplaceAll(strings.ToLower(region), "_", "-")
 }
+
+func SlicePtrToSlice[T any](v *[]T) []T {
+	if v == nil {
+		return nil
+	}
+	return *v
+}
+
+func NonEmptySliceToPtrSlice[T any](v []T) *[]T {
```

Comment:
```suggestion
func NonEmptySliceToSlicePtr[T any](v []T) *[]T {
```

nit: can we keep consistency of how to use `Ptr`? either prefix or suffix (see previous method).

Reply:
oops, thanks, changing to NonEmptySliceToSlicePtr